### ISSUE TITLE
Gromacs topology parsing: different from openmm GromacsTopFile()

### ISSUE THREAD

**Issue by EqualAPriori** [ParmEd/ParmEd#1025]
Hi there, thank you for the wonderful work, super useful stuff. New to the world of atomistic simulation -- any tips are highly appreciated!

I've been trying to simulate SDS using the gromacs-format forcefield definitions in this [recent paper](https://pubs.acs.org/doi/suppl/10.1021/acs.jctc.7b00181). 

I've tried loading the topology both using the openmm GromacsTopFile() function as well as the ParmEd parser. The openmm parser loads the topology without incident, while the ParmEd parser raises a warning 

`GromacsWarning: 6300 1-4 pairs were missing from the [ pairs ] section and were set to zero; make sure you know what you're doing!`

My interpretation is that all the 1-4 nonbonded interactions are zeroed out, which seems concerning -- 
I would've thought that these interactions should've been recognized and generated: in my .top file I include amber99.ff/forcefield.itp, which in turn has the gen-pairs and fudgeLJ, fudgeQQ terms. 

So my main questions are:
1. Is my interpretation correct? Or does this warning mean that ParmEd is missing some other interaction between the said atoms? 
2. is openMM's GromacsTopFile() also silently setting everything to zero, or using some other default behavior, different from ParmEd? Which behavior is correct?
3. is there a way to use ParmEd to turn Gromacs itp's into openMM XML definitions?

Thanks!
===== Addendum =====
I investigated what the nonbondedforce.getExceptionParameters() had to say, and can confirm that openMM's GromacsTopFile() is behaving differently than ParmEd's parser. In particular, 1-4 exceptions were non-zero in openMM's result, while ParmEd zeroed them out.

I think this is because the .itp residue definition in the link I gave above set nrexl=3 -- if I'm not mistaken, this means that ParmEd's behavior is correct, while openMM's parser is misbehaving. Could someone more knowledgeable than me could help me understand what's going on, I would highly appreciate it.

Thank you!

**Comment by swails** [comment]
Hi, sorry for the long delay.  I believe you're correct in your addendum.  I believe the way GROMACS treats exception parameters is to exclude all atom pairs within `nrexcl` bonds of each other and then add exceptions from the `[ pairs ]` section.  Most modern force fields *do* include some form of 1-4 nonbonded interactions:

* Amber force fields scale the Lennard-Jones and electrostatic interactions by a constant
* GLYCAM force fields include them fully
* CHARMM force fields include electrostatic fully and parametrize a "new" set of Lennard-Jones parameters for 1-4 partners.

Not sure what GROMOS does.  But the way I interpreted the GROMACS topology and documentation (as well as some experiments with GROMACS directly) is that if a pair of atoms is not included in the `[ pairs ]` section but is within `nrexcl` bonds from each other, their interaction is excluded completely.

However, this is unusual behavior for the force fields I'm familiar with, so I added that warning to let people know there's some weird stuff possibly happening.

> if I'm not mistaken, this means that ParmEd's behavior is correct, while openMM's parser is misbehaving.

I believe you're right here.  Only way to know for certain is to compare the energies from OpenMM and ParmEd's parsers to that produced directly by GROMACS.  One will not be like the others...
